### PR TITLE
Make fill form shortcuts optional

### LINKED ISSUE #32: [feature] get rid of X11 dependency
AFAIU the keyboard shortcut code is responsible for the dependency on X11. With desktops moving to Wayland there should be a better solution. I'm not sure if depending on X11 makes Qt use X11 instead of Wayland on a Wayland desktop.

I see three options. Please note that I haven't researched them yet, so some of them may be impossible. In order of preference:

1. figure out how to implement keyboard shortcuts using Qt APIs. If this isn't possible with Qt5, then a Qt6-only solution would be acceptable.

2. provide a Wayland implementation and determine at runtime which one to use. This wouldn't remove the X11 build dependency, but at least `qMasterPassword` wouldn't show up as X11 client at runtime.

3. provide a way to disable shortcuts at build time, f.ex. unset a flag when the build can't find X11 headers. Not nice, but better having to depend on X11.

### PR DESCRIPTION
**Pull request by stefanb2**
Fill form shortcuts are the only thing that depends on `class Keypress`, which on Linux is dependent on X11. When the program is running in a Wayland session, then those shortcuts do not work, i.e. they are useless, but still a X11 client connection is created. 

This PR makes all code related to those short cuts optional and the user can disable them with a build configuration option. I.e. she can create an executable that will be a pure Wayland client in a Wayland session.

Fixes #32

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,10 +51,14 @@ else()
     )
 endif()
 
-add_subdirectory(include)
-add_subdirectory(src)
-add_subdirectory(test)
-add_subdirectory(ui)
+if (DISABLE_FILL_FORM_SHORTCUTS)
+    message("INFO: fill form shortcuts are disabled")
```
Comment (bkueng) [inline CMakeLists.txt]:
```suggestion
    message(STATUS "fill form shortcuts are disabled")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -51,10 +51,14 @@ else()
     )
 endif()
```
Comment (bkueng) [inline CMakeLists.txt:53]: Can you add a cmake option? `option(DISABLE_FILL_FORM_SHORTCUTS "Build without form fill support" OFF)`

**Review by bkueng — COMMENTED** [review]
I'm fine with this for now but in the longer run I want to either remove it completely, or implement a wayland-compatible solution.

**Comment by stefanb2** [comment]
IMHO it should be removed, because after pondering a bit about it I feel it is a security risk to randomly try to select another window on the desktop and then insert secret information.

I still feel that `ydotool` would be a possible solution, provided it would be available as properly integrated desktop/systemd user service. But as I personally see it as a security risk I couldn't convince myself to spend the time on such an implementation.

**Comment by bkueng** [comment]
> IMHO it should be removed, because after pondering a bit about it I feel it is a security risk to randomly try to select another window on the desktop and then insert secret information.

I agree it demands a bit more from the user. But how (if at all) do you see it as more of a security risk than going through the clipboard?

**Comment by stefanb2** [comment]
> I agree it demands a bit more from the user. But how (if at all) do you see it as more of a security risk than going through the clipboard?

In the sense that CTRL-TAB selects the wrong window, f.ex. the chat window instead of the browser window with the password dialog as the user intended. Of course that maybe just me and my usual sausage fingers :-)

**Review by bkueng — APPROVED** [review]
> In the sense that CTRL-TAB selects the wrong window, f.ex. the chat window instead of the browser window with the password dialog as the user intended. Of course that maybe just me and my usual sausage fingers :-)

I'm pretty sure you could handle it :laughing:

**Pull request merged**